app/teachingCall/teachingCallStatus/controllers/ModalAddInstructorsCtrl.js

```diff
@@ -30,18 +31,49 @@ teachingCallApp.controller('ModalAddInstructorsCtrl', ['$scope', '$rootScope', '
 		'03': 'Spring Quarter'
 	};
 
-	$scope.senateGroup = angular.copy(state.eligible.senate);
-	$scope.federationGroup = angular.copy(state.eligible.federation);
-	$scope.lecturerGroup = angular.copy(state.eligible.lecturer);
-
-	$scope.startTeachingCallConfig.invitedInstructors = $scope.senateGroup.concat($scope.federationGroup);
-	$scope.startTeachingCallConfig.invitedInstructors = $scope.startTeachingCallConfig.invitedInstructors.concat($scope.lecturerGroup);
+	$scope.startTeachingCallConfig.invitedInstructors = state.calculations.instructorsEligibleForCall;
 	$scope.startTeachingCallConfig.invitedInstructors = _array_sortByProperty($scope.startTeachingCallConfig.invitedInstructors, "lastName");
 
-	$scope.startTeachingCallConfig.invitedInstructors.forEach(function(slotInstructor) {
-		slotInstructor.invited = false;
+	$scope.startTeachingCallConfig.invitedInstructors.forEach(function(instructor) {
+		instructor.invited = false;
 	});
 
+	$scope.areThereInstructorsOfType = function(instructorTypeId) {
+		var atLeastOneInstructor = false;
+
+		if ($scope.startTeachingCallConfig.invitedInstructors) {
+			$scope.startTeachingCallConfig.invitedInstructors.forEach(function(instructor) {
+				if(instructor.instructorTypeId == instructorTypeId) {
+					atLeastOneInstructor = true;
+				}
+			});
+		}
+
+		return atLeastOneInstructor;
+	};
+
+	$scope.inviteInstructorsOfType = function (instructorTypeId) {
+		$scope.startTeachingCallConfig.invitedInstructors.forEach(function(instructor) {
+			if(instructor.instructorTypeId == instructorTypeId) {
+				instructor.invited = true;
+			}
+		});
+
+		$scope.startTeachingCallConfig.isAddInstructorFormComplete = $scope.isAddInstructorFormComplete();
+	};
+
+	$scope.areAllInstructorsOfTypeInvited = function (instructorTypeId) {
+		var areAllInvited = true;
+
+		$scope.startTeachingCallConfig.invitedInstructors.forEach(function(instructor) {
```

Comment:
Use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find and get rid of the temporary variable